source/_posts/2018-08-06-Android WorkManager.md

```diff
@@ -0,0 +1,128 @@
+# Android WorkManager
+
+## Introduction
+
+`WorkManager` is a new API in Android Architecture Components introduced in the Google I/O 2018. It simplifies and makes it much easier to do work on background threads. The `WorkManager` schedules tasks as instances of the `Worker` class and can schedule these workers based on certain conditions, which you can set by using the provided The `Constraints` class. Examples of conditions you can set from the `Constraints` class, can be things like available internet/wifi connection or if a charger is connected. The `WorkManager` can also schedule all `Worker` instance you have to launch in any order you wish.
+</br>
+</br>Also a very important note about [`WorkManager`](https://developer.android.com/topic/libraries/architecture/workmanager): </br> *“WorkManager is intended for tasks that require a guarantee that the system will run them even if the app exits...”*
+
+
+## How to use it
+
+First you have to add the dependency in your `build.gradle` file: </br>
+`implementation "android.arch.work:work-runtime:1.0.0-alpha02"`
+
+To demonstrate how to use `WorkManager` we will start by establishing a made-up user story.
+In our imaginary app QuickSnapper, we press a shutter button and the app will take a picture and apply some stickers on it and upload it all in one automatic process thanks to the WorkerManager.
+
+So let's split up the user story in 3 use cases:
+
+- 1 The user takes a photo (We want to compress it here) 
+- 2 The app adds weather and location information on the picture after step 1 (GPS and Internet must be available)
+- 3 The app uploads the image immediately after step 1-2  (Internet must be available) 
+
+
+For each use case we will make a `Worker` class. To do that we need to make a class and extend `Worker` which requires us to implement a `doWork()` method with a return type of a `WorkerResult` that can either be `WorkerResult.SUCCESS` or `WorkerResult.FAILURE`
+
+#### 1) Here is the first Worker which compress our image into a smaller size and returns either WorkerResult.SUCCESS or WorkerResult.FAILURE
+
+```java
+ class ImageCompressionTask extends Worker {
+
+   private Bitmap bitmap;
+
+   public ImageCompressionTask(Bitmap bitmap) {
+       this.bitmap = bitmap;
+   }
+
+   @NonNull
+   @Override
+   public WorkerResult doWork() {
+       Bitmap newBitmap;
+       try {
+           newBitmap = Bitmap.createScaledBitmap(bitmap, 500, 500, false);
```

Comment:
shouldn't we do something with the bitmap? I know it's proof of concept, but still :D

Reply:
At least add in a comment with `// Write new bitmap to persistence` or something